### ISSUE TITLE
Warn on unused chain result

### ISSUE THREAD

**Issue by KingOfBrian** [Raizlabs/BonMot#133]
This may not be possible with the current obj-c implementation, but if you build a chain, ideally the compiler would warn you if the new result is unused.  This is to protect against:

```
        let chain = BONChain()
        chain.lineSpacing(1).textColor(UIColor.blueColor())
        return chain
```

I tried adding `__attribute__((warn_unused_result))` to the block definitions, but it failed with:
``` 
'warn_unused_result' attribute only applies to functions, methods, and classes
```

**Comment by ZevEisenberg** [comment]
Can you add it to the properties?

**Comment by ZevEisenberg** [comment]
It doesn't look like this is possible on Obj-C (properties are a no-go because you do "use" the property by invoking the block it returns) or Swift (can't get it to accept `@warn_unused_result` on the closures or properties). Leaving this open in case things change in the future.

**Comment by KingOfBrian** [comment]
Resolved in 4.0
